### ISSUE TITLE
MenuItems should not have any focus ring

### ISSUE THREAD

**Issue by herhuey** [opengovsg/design-system#244]
**Describe the bug**
Onhover shows focus ring in some instances and no focus ring in other instances (view screenshots). 

From Kar Rui: focus only shows using the :focus-visible selector, which may have been triggered when you open (since programatic focus === focus-visible)

**To Reproduce**
1. Go to https://design.hack.gov.sg/?path=/docs/components-menu--docs
2. Hover over any MenuItem and note the current focus ring behaviour
3. Repeatedly refresh page and hover over any MenuItems until the focus ring behaviour changes

**Expected behavior**
MenuItems should not show any focus ring.

**Screenshots**
![image](https://user-images.githubusercontent.com/122253703/219554001-83088149-92ea-426d-aa45-59da536e72c1.png)
![image](https://user-images.githubusercontent.com/122253703/219554013-a7b0f3da-7dd2-462f-af9f-e9815fca20c6.png)

**Desktop (please complete the following information):**
 - OS: macOS Ventura 13.1
 - Browser: Chrome
 - Version 109.0.5414.119 (Official Build) (x86_64)

**Comment by seaerchin** [comment]
i found that this bug could be reliably triggered by using `tab` (on keyboard) then after, using mouse to hover triggers the focus ring
